### ISSUE TITLE
[Feature request] Add support for FireTV

### ISSUE THREAD

**Issue by adwaitv** [TeamNewPipe/NewPipe#899]
Google and Amazon are not being nice to each other. YouTube will be blocked on Fire TV on new year's day. 

Let's not let the consumer suffer while the companies are bickering. Let's bring newpipe to fire TV

**Comment by wb9688** [comment]
If Fire TV is just Android TV, I'm working on it. If not, you're likely out
of luck.

**Comment by theScrabi** [comment]
Fire tv is android as long as i know

**Comment by wb9688** [comment]
@theScrabi: But Android TV has a totally different 'Leanback' interface and
I'm not sure if Fire TV uses standard Leanback or an own thing.

**Comment by theScrabi** [comment]
We may have to try it. I found this documebtation here: https://developer.amazon.com/docs/fire-tv/differences-from-android-tv-development.html

**Comment by jpf91** [comment]
Actually, NewPipe can be installed on FireTV already and it works more or less fine (of course, the interface could always be optimized for remote control usage, but it can e used right now).

The main problem for now is when activating the search bar, for some reason the on-screen-keyboard does not appear. You can use an external keyboard, but most people probably don't have a bluetooth keyboard for the FireTV stick.

**Comment by theScrabi** [comment]
Maybe that was also a good idea to just get the current ui to work on fireTV until the leanback UI is ready.

**Comment by geebotron** [comment]
@jpf91 Yes I agree. The only thing preventing NewPipe being fully usable with the Fire remote is the fact that the on-screen keyboard doesn't pop-up when the text entry box for the search is selected / focused.

**Comment by winnebagel** [comment]
@wb9688 I have put newpipe on a fire tv, the issue is that it isn't at all remote friendly.
I'm not at all knowledgeable of code, but you can flash the current apk onto a fire tv.
The apk works, just not well with remote.

**Comment by ChairmanMkay** [comment]
Of the functions I have tested the most problematic is emailing a crash report. There is no email client available. I'm researching a side load option. Popup video always crashes on my 4k stick.  I am using a USB ethernet bridge for throughput.

I had hoped to watch (skim) 8 daily hours of senate trial video. After a frustrating 9 minutes of searching the video does play full screen but the aspect ratio is extremely wonky. It fills the screen width but the height is not commensurate. I selected 720p video.
